src-electron/generator/matter/darwin/Framework/CHIP/templates/helper.js

```diff
@@ -319,7 +319,11 @@ function objCEnumName(clusterName, enumLabel, options) {
   return 'MTR' + clusterName + enumLabel;
 }
 
-function objCEnumItemLabel(itemLabel) {
+function objCEnumItemLabel(itemLabel, options) {
+  if (options.hash.preserveAcronyms) {
+    return appHelper.asUpperCamelCase.call(this, itemLabel, options);
+  }
```

Comment:
I am a bit confused by this part. Why don't you use `asUpperCamelCase` in the template directly ?

Reply:
This would allow me to pass the preserveAcronyms boolean to `objCEnumItemLabel` from other parts of the template if needed. But yes, it's possible we can just do `asUpperCamelCase` directly, with just the fixes to it above.